### ISSUE TITLE
Rails converts Time column to DateTime on select

### ISSUE THREAD

**Issue by davidcompass** [rails/rails#43629]
### Steps to reproduce

From my rails console

```ruby
3.0.2 :004 > AppointmentBlock.columns_hash["starttime"].type
 => :time 
3.0.2 :005 > AppointmentBlock.pluck(:starttime)
   (31.2ms)  SELECT `appointment_blocks`.`starttime` FROM `appointment_blocks`
 => 
[Sat, 01 Jan 2000 07:00:00.000000000 CET +01:00,
 Sat, 01 Jan 2000 07:00:00.000000000 CET +01:00,
 Sat, 01 Jan 2000 07:00:00.000000000 CET +01:00,
 Sat, 01 Jan 2000 07:00:00.000000000 CET +01:00,
 Sat, 01 Jan 2000 07:00:00.000000000 CET +01:00,

```

### Expected behavior
It should return a Time type object

### Actual behavior
Returns a DateTime

### System configuration
**Rails version**:
6.1.4.1

**Ruby version**:
3.0.2

**MySql Version**:
5.6.36

**Comment by jacobherrington** [comment]
Hey @davidcompass, I took some time to reproduce this error (getting MySQL 5.6 running on my laptop wasn't as easy as I hoped).

I've added a reproduction script to the bottom of this comment (or at least, what I think it'd look like to reproduce what you're describing).

As far as I can tell, I'm not able to reproduce this issue in a test. Is this something you're still encountering? Let me know if you can share any additional information explaining why we are getting different results.

You could try running this script and see if you have a different result than me as well.

OS:
Max OS Big Sur 11.6

Rails version:
6.1.4.1

Ruby version:
3.0.2

MySql version:
5.6.51 (slight variation here)

mysql2 version:
0.5.3

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "rails", "6.1.4.1"
  gem "mysql2"
end

require "active_record"
require "minitest/autorun"
require "logger"

ActiveRecord::Base.establish_connection(
  adapter: "mysql2",
  user: "root",
  database: "time_test"
)
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.time :timestamp
  end
end

class Post < ActiveRecord::Base; end

class BugTest < Minitest::Test
  def test_stuff
    5.times do
      Post.create! timestamp: Time.now + rand(1..5).hours
    end

    assert_equal :time, Post.columns_hash["timestamp"].type
    assert_equal Time, Post.first.timestamp.class
    assert_equal [Time], Post.pluck(:timestamp).map(&:class).uniq!

    puts Post.pluck(:timestamp)
    #=> 2000-01-01 01:40:30 UTC
    #=> 2000-01-01 23:40:30 UTC
    #=> 2000-01-01 01:40:30 UTC
    #=> ...
  end
end
```

**Comment by davidcompass** [comment]
hey @jacobherrington

First of all thank you very much for taking the time to look at this. I ran your script and came up with the same results.
I understand now and it seems like I got confused distinguishing the time type and the datetime. Because my problem is actually that the time includes the Date of 2000-01-01.

`#=> 2000-01-01 01:40:30 UTC`

This statement here used to work before `ruby 1.9.3, rails 3.2` but now returns `nil`

`AppointmentBlock.select("round(time_to_sec(min(starttime))/60)-#{start_buffer}-#{slot_size} as starttime").where('appointment_maintype_id = 1').first.starttime.to_i`

If I execute the resulting statement:
`SELECT round(time_to_sec(min(starttime))/60)-60-20 as starttime FROM appointment_blocks WHERE (appointment_maintype_id = 1) ORDER BY appointment_blocks.id`

The result is what I expect. Just a number for example 460.

Somewhere the executing "breaks" as in returning `nil`. I am actually not sure at all what is the issue here and it might have nothing to do with rails. 

If you have any idea how to make this work again I would be grateful. 
I already tried to force this with an Arel.sql wrapper but that doesn't seem to work.
